### PR TITLE
Audit fixes

### PR DESCRIPTION
**Pull request by omartijn**
This PR implements the following suggestions given by our code auditors:

- allow specifying preferred symmetric key, hash and compression algorithms
- secure allocator and wrapper object for objects using sensitive data

By making a contribution to this project, I certify that:

        (a) The contribution was created in whole or in part by me and I
            have the right to submit it under the GPL-3.0 license; or

        (b) The contribution is based upon previous work that, to the best
            of my knowledge, is covered under an appropriate open source
            license and I have the right under that license to submit that
            work with modifications, whether created in whole or in part
            by me, under GPL-3.0 license; or

        (c) The contribution was provided directly to me by some other
            person who certified (a), (b) or (c) and I have not modified
            it.

        (d) I understand and agree that this project and the contribution
            are public and that a record of the contribution (including all
            personal information I submit with it, including my sign-off) is
            maintained indefinitely and may be redistributed consistent with
            this project or the license(s) involved.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,7 +147,7 @@ int main()
     // a range_encoder around it. the range_encoder works by writing
     // the raw data to a provided range of bytes, which must stay in
     // scope during the encoder operation.
-    std::vector<uint8_t> data(packet.size());
+    pgp::vector<uint8_t> data(packet.size());
```
Comment (bysnack) [inline README.md]: Update to match the implementation

File: README.md [hunk 2]
```diff
@@ -182,8 +186,8 @@ int main()
     // need an extra byte in front of it, it is always the same byte
     // for this key type, so it doesn't add any real information but
     // pgp still requires it and throws a fit if it is missing.
-    std::vector<uint8_t> public_key_data(crypto_sign_PUBLICKEYBYTES + 1);
-    std::vector<uint8_t> secret_key_data(crypto_sign_SECRETKEYBYTES);
+    pgp::vector<uint8_t> public_key_data(crypto_sign_PUBLICKEYBYTES + 1);
```
Comment (bysnack) [inline README.md]: Update to match the implementation

File: README.md [hunk 3]
```diff
@@ -246,7 +250,7 @@ int main()
 
     // we now have a set of packets, which, when encoded to a file, can
     // be imported into a compatible pgp implementation (such as gnupg)
-    std::vector<uint8_t> data(
+    pgp::vector<uint8_t> data(
```
Comment (bysnack) [inline README.md]: Update to match the implementation

**Comment by omartijn** [comment]
Well found about the reserved names. I read somewhere else that it only applies to the global namespace - which isn't what we're doing here, so I am unsure if it could even theoretically be a problem. I'd say let's leave it for now and properly evaluate whether or not we need to change this, because I have used this in other places as well.

**Pull request merged**